src/Raven.Server/Documents/Handlers/AttachmentHandler.cs

```diff
@@ -0,0 +1,207 @@
+            }
+        }
+
+        [RavenAction("/databases/*/attachments", "PUT")]
+        public async Task Put()
+        {
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            {
+                var id = GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
+                var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+                var contentType = GetStringQueryString("contentType", false) ?? "";
+
+                var tempPath = GetUniqueTempFileName(Path.GetTempPath(), "tmp");
+                using (var file = new FileStream(tempPath, FileMode.CreateNew, FileAccess.ReadWrite, FileShare.None, 4096, FileOptions.DeleteOnClose) )
+                {
+                    await RequestBodyStream().CopyToAsync(file, 81920, Database.DatabaseShutdown);
+                    file.Position = 0;
+
+                    var etag = GetLongFromHeaders("If-Match");
+
+                    var cmd = new MergedPutAttachmentCommand
+                    {
+                        Database = Database,
+                        ExpectedEtag = etag,
+                        DocumentId = id,
+                        Name = name,
+                        Stream = file,
+                        ContentType = contentType,
+                    };
+                    await Database.TxMerger.Enqueue(cmd);
+                    cmd.ExceptionDispatchInfo?.Throw();
+
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
+
+                    using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                    {
+                        writer.WriteStartObject();
+
+                        writer.WritePropertyName(nameof(PutResult.ETag));
+                        writer.WriteInteger(cmd.EtagResult);
+
+                        writer.WriteEndObject();
+                    }
+                }
+            }
+        }
+
+        [RavenAction("/databases/*/attachments", "DELETE")]
+        public async Task Delete()
+        {
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            {
+                var id = GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
+                var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+
+                var etag = GetLongFromHeaders("If-Match");
+
+                var cmd = new MergedDeleteAttachmentCommand
+                {
+                    Database = Database,
+                    ExpectedEtag = etag,
+                    DocumentId = id,
+                    Name = name,
+                };
+                await Database.TxMerger.Enqueue(cmd);
+                cmd.ExceptionDispatchInfo?.Throw();
+
+                NoContentStatus();
+            }
+        }
+
+        const string TempFilePrefix = "RavenDB.";
```

Comment:
RavenDB is incredibly generic term for us.  And why do we need / want to have a meaningful name here?
Call it `attachment-{Guid.NewGuid()}` and be done with it.